### PR TITLE
README: Add "How to backport changes to a specific branch ?"

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,33 @@ where:
 For more details, see https://www.slicer.org/wiki/Documentation/Nightly/Developers/ProjectForks
 
 
+How to backport changes to a specific branch ?
+----------------------------------------------
+
+1. Fork `Slicer/VTK`
+
+2. Checkout the relevant `slicer-vX.Y.Z-YYYY-MM-DD-SHA{7}` branch
```
Comment (lassoan) [inline README.md]: How do we know which one is the "relevant" branch? The one that is currently used by the superbuild (https://github.com/Slicer/Slicer/blob/master/SuperBuild/External_VTK.cmake#L129)?

File: README.md [hunk 2]
```diff
@@ -25,6 +25,33 @@ where:
...
+
+```
+[backport MR-6941] Fix infinite loop in vtkContourTriangulator 
+
+Cherry-picked commit a93fff6 from main VTK repository.
+
+...
+```
+
+6. Finally, share your updated branch with the Slicer development team on discourse
```
Comment (lassoan) [inline README.md]: It would be simpler to just submit a pull request. Any questions can be discussed there with maintainers.

**Review by lassoan — COMMENTED** [review]
Thanks a lot, this clarifies a lot and the described process is much simpler. I'm wondering if we should describe the typical scenario that happens during Slicer development: user starts from a Slicer build tree, so he just needs to add kitware/VTK as a remote, cherry pick commits, update comment, push it to his fork, and send a pull request to the current Slicer/VTK branch.

**Comment by jcfr** [comment]
@lassoan This should address your comments.

I am also wondering if adding a section to slicer developer guide on readthedocs and referencing it would be more helpful.

**Comment by lassoan** [comment]
If the exact same process applied to many repositories then it would make sense to describe it in one place, in the developer guide.

Can you explain why is it necessary to change the commit comments?

**Comment by jcfr** [comment]
> Can you explain why is it necessary to change the commit comments?

Done

**Comment by lassoan** [comment]
How commit comments help better than git rebase, which can track which commits have been merged already and can skip those? Do you find that git rebase is not reliable (due to merge conflicts, ...)?

What about marking Slicer-specific changes very visibly instead? There should be much less (if any) of those.

**Comment by jcfr** [comment]
> How commit comments help better than git rebase, which can track which commits have been merged already and can skip those?

In practice,  git rebase has not very helpful because cherry-picked commit are sometimes modified in subtle way. Cherry-picking them one by one (or in small batch specifying a range) has been more productive.

Also since (1) version of commit associated with merge request may change in the upstream MR or PR after they are cherry-picked and (2) the commit title may also change during review in upstream repo, only having the commit message is not enough. Indicating the MR or PR number is very helpful.

Being as explicit as possible to indicate the provenance of the commit has helped me a lot.

> What about marking Slicer-specific changes very visibly instead? There should be much less (if any) of those.

That would also work, that said being explicit turns out to be helpful and not a burden.

I am thinking we should probably do both:

```
[backport] ... 
[backport-MR-12324] ... 
[Slicer] ...
```

**Comment by lassoan** [comment]
OK, I see. I understand now why we cannot rely only on rebase to manage these patches.

**Pull request merged**